### ISSUE TITLE
How to stabilize NaiveBayesClassifier outputs?

### ISSUE THREAD

**Issue by alvations** [nltk/nltk#2097]
On sometimes we see that the continuous integration test fails because of some non-deterministic results from the `NgramTagger` i.e. are `NaiveBayesClassifer`, e.g. https://nltk.ci.cloudbees.com/job/pull_request_tests/822/PYV=3.6.4,jdk=jdk8latestOnlineInstall/testReport/junit/(root)/nltk/tag/

Is there someway to set a random seed or something such that classifier's output is consistent and make sure that the model is easily replicable?

**Comment by iliakur** [comment]
Yes, there is [a way to seed randomness](https://docs.python.org/3/library/random.html#random.seed)!

I would also recommend using a seeded `random.Random` so that setting a seed in one part of NLTK doesn't affect any other parts.

Actually, an example of how this works can be found in `LanguageModel.generate`, [here](https://github.com/nltk/nltk/blob/develop/nltk/lm/api.py#L196)

**Comment by purificant** [comment]
Which test is flaky? Some links in the above comments are not longer valid.

**Comment by iliakur** [comment]
@purificant very sorry, I dunno how that disney link made it into my comment. I have updated it to only contain relevant python links!

**Comment by purificant** [comment]
Thanks @Copper-Head! It was fun to read :)

@alvations I don't suppose you know which test is flaky? Since CI link is no longer accessible. I've not seen any flaky tests myself on Travis or local tox.

**Comment by alvations** [comment]
The test would still fail at times for sure if we don't do something to the non-deterministic output checks. I'll dig the specific doctest or unittest up

**Comment by andvea** [comment]
Hi @alvations and @iliakur!
Is this still open? If so, may I have more context regarding the non-deterministic results of `NaiveBayesClassifer` and the mentioned tests that sometimes fail?
Thank you!

**Comment by iliakur** [comment]
@andvea sorry, I don't actually know what was up with `NaiveBayesClassifier`. I only responded with an example for how to seed randomness ;)
